crates/extensions/c8y_mapper_ext/src/operations/convert.rs

```diff
@@ -387,4 +390,42 @@ impl CumulocityConverter {
             Ok(messages) => Ok(messages),
         }
     }
+
+    pub fn convert_custom_operation_request(
+        &self,
+        device_xid: String,
+        cmd_id: String,
+        command_name: String,
+        custom_handler: &Operation,
+        message: &MqttMessage,
+    ) -> Result<Vec<MqttMessage>, CumulocityMapperError> {
+        let entity_xid: EntityExternalId = device_xid.into();
+
+        let target = self.entity_store.try_get_by_external_id(&entity_xid)?;
+
+        let channel = Channel::Command {
+            operation: OperationType::Custom(command_name),
+            cmd_id,
+        };
+
+        let topic = self.mqtt_schema.topic_for(&target.topic_id, &channel);
+
+        let state = GenericCommandState::from_command_message(message).map_err(|e| {
+            CumulocityMapperError::JsonCustomOperationHandlerError {
+                operation: custom_handler.name.clone(),
+                err_msg: format!("Invalid JSON message, {e}. Message: {message:?}"),
+            }
+        })?;
+
+        let payload: Value = if let Some(tedge_cmd_value) = custom_handler.tedge_cmd_payload() {
+            let payload = state.inject_values_into_template(tedge_cmd_value);
+            serde_json::from_str(&payload)?
+        } else {
+            serde_json::Value::Null
+        };
```

Comment:
This can be improved using the same logic as for spawning a sub-operation using [`StateExcerpt`](https://github.com/thin-edge/thin-edge.io/blob/main/crates/core/tedge_api/src/workflow/state.rs#L540)

```
let excerpt = StateExcerpt::PathExpr(custom_handler.tedge_cmd_payload() );
let payload = excerpt.extract_value_from(state);
```

The main benefit is that this will help to compatible with workflow definition and enable feature like building the tedge operation payload as a *combination* of several fragments of the c8y operation payload (as done [here](https://thin-edge.github.io/thin-edge.io/references/agent/operation-workflow/#sub-operation-execution):

```
command_payload.x = "some static value"
command_payload.y = "${.payload.y}"
command_payload.z = "${.payload.z}"
```

Reply:
I'm okay to merge this PR without this refinement.

However, we need to agree on the property names as discussed [here](https://github.com/thin-edge/thin-edge.io/issues/3095#issuecomment-2473876874) and [here](https://github.com/thin-edge/thin-edge.io/issues/3095#issuecomment-2490426004). @reubenmiller what's your point of view of these property names?

Reply:
Ok, I've added my suggestion which is merging the two concepts. https://github.com/thin-edge/thin-edge.io/issues/3095#issuecomment-2493486901